pulp_container/app/registry_api.py

```diff
@@ -473,6 +477,135 @@ def get_queryset(self, *args, **kwargs):
+
+        results = []
+        for distribution in distributions:
+            images = []
+            if distribution.repository:
+                repository_version = distribution.repository.latest_version()
+            else:
+                repository_version = distribution.repository_version
+            tags = models.Tag.objects.filter(pk__in=repository_version.content)
+            if req_tags:
+                tags = tags.filter(name__in=req_tags)
+            manifests = {}  # mapping manifests to sets of encountered tag names
+            for tag in tags:
+                self.recurse_through_manifest_lists(
+                    tag.name, tag.tagged_manifest, req_oss, req_architectures, manifests
+                )
+            for manifest, tagged in manifests.items():
+                with storage.open(manifest.config_blob._artifacts.get().file.name) as file:
+                    raw_data = file.read()
+                config_data = json.loads(raw_data)
+                labels = config_data.get("config", {}).get("Labels")
+                if not labels:
+                    continue
+                if any(label not in labels.keys() for label in req_label_exists):
+                    continue
+                os = config_data["os"]
+                if req_oss and os not in req_oss:
+                    continue
+                architecture = config_data["architecture"]
+                if req_architectures and architecture not in req_architectures:
+                    continue
+                if any(
+                    labels.get(label) not in values for label, values in req_label_values.items()
+                ):
+                    continue
+                if not req_tags:
+                    direct_tags = models.Tag.objects.filter(
+                        tagged_manifest__pk=manifest.pk
+                    ).values_list("name", flat=True)
+                    tagged.update(direct_tags)
+                images.append(
+                    {
+                        "Tags": tagged,
+                        "Digest": manifest.digest,
+                        "MediaType": manifest.media_type,
+                        "OS": os,
+                        "Architecture": architecture,
+                        "Labels": labels,
+                    }
+                )
+            if images:
+                results.append({"Name": distribution.base_path, "Images": images})
+
+        return Response(data={"Registry": settings.CONTENT_ORIGIN, "Results": results})
+
+
+class FlatpakIndexStaticView(FlatpakIndexDynamicView):
+    """
+    Handles requests to the /index/static endpoint
+    """
+
+    @FlatpakIndexStaticCache()
+    def get(self, request):
+        response = super().get(request)
+        response.accepted_renderer = JSONRenderer()
+        response.accepted_media_type = JSONRenderer.media_type
+        response.renderer_context = {}
+        response.render()
```

Comment:
Is this necessary to do?

Reply:
yes, added comment in c6605fe350000eaa3c0de35d05f73af25453405d